agent/exec/containerd/executor.go

```diff
@@ -15,53 +13,29 @@ import (
 	"github.com/docker/swarmkit/log"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/grpclog"
 )
 
 type executor struct {
-	conn             *grpc.ClientConn
+	client           *containerd.Client
 	secrets          exec.SecretsManager
 	genericResources []*api.GenericResource
-	containerDir     string
 }
 
 var _ exec.Executor = &executor{}
 
-func getGRPCConnection(ctx context.Context, sock string) (*grpc.ClientConn, error) {
-	grpclog.SetLogger(log.G(ctx))
-
-	dialOpts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithTimeout(100 * time.Second)}
-	dialOpts = append(dialOpts,
-		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
-			return net.DialTimeout("unix", sock, timeout)
-		},
-		))
-
-	conn, err := grpc.Dial("unix://"+sock, dialOpts...)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to dial %q", sock)
-	}
-
-	return conn, nil
-}
-
 // NewExecutor returns an executor using the given containerd control socket
-func NewExecutor(sock, stateDir string, genericResources []*api.GenericResource) (exec.Executor, error) {
-	ctx := log.WithModule(context.Background(), "containerd")
-
-	conn, err := getGRPCConnection(ctx, sock)
+func NewExecutor(sock string, genericResources []*api.GenericResource) (exec.Executor, error) {
+	// TODO(ijc), configurable namespace.
+	// TODO(ijc), default to swarmd?
+	client, err := containerd.New(sock, containerd.WithDefaultNamespace("default"))
```

Comment:
I tend to think of `swarmd` the standalone daemon process as somewhat separate to the `swarmkit` toolkit. So I think I might take a two layered approach here and have `cmd/swarmd/main.go` define a `--containerd-namespace` option, defaulting to `"swarmd"` which is passed as an argument to `NewExecutor` here.

In `NewExecutor` if the requested namespace is `""` then it'll default to `swarmkit`. Or maybe I'll make it an error.